### PR TITLE
Adds Vue.js instruction tab on no data screen

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added site detection for Vue and added relevant tab and info.
Fixes: #PG-2778

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by sgiehl — APPROVED** [review]

**Comment by sgiehl** [comment]
Looks good. Will wait for the tests to pass...

File: tests/PHPUnit/Fixtures/EmptySiteWithSiteContentDetectionVue.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            'ga3' => false,
+            'ga4' => false,
+            'gtm' => false,
+            'cms' => SitesManager::SITE_TYPE_UNKNOWN,
+            'cloudflare' => false,
+            'jsFramework' => SitesManager::JS_FRAMEWORK_VUE,
+        ];
+
+        return [
+            SiteContentDetector::class => \Piwik\DI::autowire(FakeSiteContentDetector::class)
```
Comment (sgiehl) [inline tests/PHPUnit/Fixtures/EmptySiteWithSiteContentDetectionVue.php]:
```suggestion
            SiteContentDetector::class => \DI::autowire(FakeSiteContentDetector::class)
```

File: tests/PHPUnit/Fixtures/EmptySiteWithSiteContentDetectionVue.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            'ga3' => false,
+            'ga4' => false,
+            'gtm' => false,
+            'cms' => SitesManager::SITE_TYPE_UNKNOWN,
+            'cloudflare' => false,
+            'jsFramework' => SitesManager::JS_FRAMEWORK_VUE,
+        ];
+
+        return [
+            SiteContentDetector::class => \DI::autowire(FakeSiteContentDetector::class)
```
Comment (sgiehl) [inline tests/PHPUnit/Fixtures/EmptySiteWithSiteContentDetectionVue.php]:
```suggestion
            SiteContentDetector::class => \DI\autowire(FakeSiteContentDetector::class)
```

Comment (AltamashShaikh) [reply tests/PHPUnit/Fixtures/EmptySiteWithSiteContentDetectionVue.php]: Thanks @sgiehl :)

**Pull request merged**